src/azure-cli/setup.py

```diff
@@ -149,7 +149,7 @@
     'semver==2.13.0',
     'six>=1.10.0',  # six is still used by countless extensions
     'sshtunnel~=0.1.4',
-    'urllib3[secure]',
+    'urllib3',
```

Comment:
`urllib3[secure]` installs below packages: 

https://github.com/urllib3/urllib3/blob/e67f13c44fa4aada1954df96eb8cccbb5f2dbd2a/pyproject.toml#L49-L55

```toml
secure = [
  "pyOpenSSL>=17.1.0",
  "cryptography>=1.9",
  "idna>=2.0.0",
  "certifi",
  "urllib3-secure-extra",
]
```

We do explicitly require them in our `requirements.txt`: https://github.com/Azure/azure-cli/blob/424374558bedf9031069e5a70e712dbe79a2442c/src/azure-cli/requirements.py3.Linux.txt

This complies with the migration guidance: https://github.com/urllib3/urllib3/issues/2700

>   * If needed, add `pyOpenSSL>=0.14`, `cryptography>=1.3.4`, `idna>=2`, and `certifi` to their dependencies. These dependencies should only be added back if they're actually used within the project. If they're not used (including `pyopenssl.inject_into_urllib3`) then they can likely be omitted.

Besides, according to https://urllib3.readthedocs.io/en/stable/reference/contrib/pyopenssl.html

> However, pyOpenSSL depends on cryptography, which depends on idna, so while we use all three directly here we end up having relatively few packages required.